lib/Dialect/FIRRTL/FIRRTLOpInterfaces.cpp

```diff
@@ -74,6 +74,16 @@ LogicalResult circt::firrtl::verifyModuleLikeOpInterface(FModuleLike module) {
           "annotations must be dictionaries or subannotations");
   }
 
+  // Verify the port symbols.
+  auto portSymbols = module.getPortSymbolsAttr();
+  if (!portSymbols)
```

Comment:
at least right now, in hw, an empty array and an empty array are valid no-symbols representations.

Reply:
Good point, it would probably make sense to reflect what `portAnnotations` does above, and the HW dialect, accepting empty lists as valid no-symbols placeholder.

Reply:
It looks like there's some additional cleanup on FIRRTL port annotations that are necessary that follow this pattern. I'll do a follow-up PR for those.